x-pack/plugins/endpoint/public/applications/endpoint/view/app_root_provider.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, ReactNode, useMemo } from 'react';
+import { Provider } from 'react-redux';
+import { I18nProvider } from '@kbn/i18n/react';
+import { Router } from 'react-router-dom';
+import { History } from 'history';
+import { CoreStart } from 'kibana/public';
+import { useObservable } from 'react-use';
+import { EuiThemeProvider } from '../../../../../../legacy/common/eui_styled_components';
+import { KibanaContextProvider } from '../../../../../../../src/plugins/kibana_react/public';
+import { appStoreFactory } from '../store';
+import { RouteCapture } from './route_capture';
+import { EndpointPluginStartDependencies } from '../../../plugin';
+
+/**
+ * Provides the context for rendering the endpoint app
+ */
+export const AppRootProvider = memo<{
```

Comment:
New - as mentioned in comment above. Will enable us to reuse the endpoint application runtime context providers in test cases.